### ISSUE TITLE
Feature request: an option for thumbnails to retain their origin's aspect ratio

### ISSUE THREAD

**Issue by Dreomite** [rr-/szurubooru#351]
The current thumbnail layout does match the overall szurubooru style, but it has one major drawback: too much important information is being cut out.

With lots of wide / tall drawings, comic strips, manga panels the preview page becomes a bit unreadable. Not only it makes all the images the same indistinguishable size, it also frequently cuts out crucial image parts making finding wanted pieces even harder.

# Proposed solutions

## 1. The old-school booru style

Retain classic booru positioning `[N x M]` (just like it is right now), but instead of stretching generated thumbnail to fill preview cells, fit it inside rectangles with set `post_height` and `post_width` sides.

![image; the author's Tripdancer by the way](https://user-images.githubusercontent.com/35058624/94270980-9e321800-ff49-11ea-87e2-52e979ba17ec.png)

Drawbacks: ugly, wastes a lot of space.

## 2. Content-aware layout

A very nice example of this approach is fairly recent [DeviantArt](https://www.deviantart.com/) redesign.

![image](https://user-images.githubusercontent.com/35058624/94287832-751c8200-ff5f-11ea-8860-303a735a8a42.png)

In my opinion, it not only makes layout being more readable, but it also looks slicker.

It's much easier to look for an inspiration this way, find needed pieces faster, and maybe even read contents of an image without opening it.

Another example is Piwigo with ['modus' extension](https://github.com/Piwigo/piwigo-modus). Here's [\[demo\]](https://piwigo.org/demo/). Though, it's not as good as on DeviantArt because it can fail in certain situations when you play with page's zoom.

Unfortunatelly, you cannot do this in pure CSS as it requires some JS to manually calculate thumbnails' width and height. However, you can come pretty close if you change following things in `client/css/post-list-view.styl`:
```css
.post-list ul li 
  flex-grow: 1; /* Comment out */
  min-width: 10em; /* Comment out */
  width: 12vw; /* Comment out */

.post-list ul li:not(.flexbox-dummy)
  height: 9vw; /* -> 14vw */

.post-list ul li .thumbnail-wrapper.no-tags .thumbnail
  outline: 4px solid #f44; /* -> 2px */

.post-list ul li &:hover a, a:active, a:focus .thumbnail
  outline: 4px solid #24aadd !important; /* -> 2px */

.post-list ul li .thumbnail
  outline-offset: -3px; /* -> -1px */
```

This way nothing will be cut out, but there will be margin on the right. If you leave `flex-grow: 1;` intact, it will evenly stretch preview cells by cutting some piece of a thumbnail which is still much better than what szurubooru currently does. I think it looks very good on mobile as well.

Here's how it looks in my browser (with `flex-grow: 1;` intact, hence some previews are cropped):
![image](https://user-images.githubusercontent.com/35058624/94295933-b2d2d800-ff6a-11ea-8a71-90486c05c6df.png)

Here's how it looks on my phone:
![image](https://user-images.githubusercontent.com/35058624/94296043-e0b81c80-ff6a-11ea-88fc-9f1989e0cc5d.png)

EDIT: I should mention that DeviantArt does cut long pictures, but it does a pretty good job at it. [An example](https://www.deviantart.com/randowis/gallery/57230689/we-live-in-an-m-m-o-comic-series).

**Comment by sgsunder** [comment]
The content aware layout looks really good! I'll try to write a PR to make it a user toggle option.

> Unfortunatelly, you cannot do this in pure CSS as it requires some JS to manually calculate thumbnails' width and height.

I think it's fine to use JS for this, as long as it remains somewhat lightweight.

**Comment by sgsunder** [comment]
Can you check out PR #352 and let me know if you think there are any changes that should be done?

**Comment by Dreomite** [comment]
That was fast. Thank you!

On my PC it seems to behave the exact same way it behaved on my local branch. But on my smartphone, if feel like some thumbnails are being cut way too aggressively.

Same image set for comparison:
![Screenshot_20200927-223045_Fennec_F-Droid](https://user-images.githubusercontent.com/35058624/94374005-dd12ca00-0111-11eb-8a02-e8a81270543e.png)

**Comment by Dreomite** [comment]
Adding `min-width: inherit` alongside `width: inherit` seems to do the job.

**Comment by sgsunder** [comment]
Implemented in eaa6107a6ca6cca7f67802b9bf939f662f9343d5
